nilearn/signal.py

```diff
@@ -508,11 +517,28 @@ def clean(signals, sessions=None, detrend=True, standardize=True,
     signals = _ensure_float(signals)
     signals = _standardize(signals, normalize=False, detrend=detrend)
 
+    # Apply low- and high-pass filters
+    if low_pass is not None or high_pass is not None:
+        if t_r is None:
+            raise ValueError("Repetition time (t_r) must be specified for "
+                             "filtering")
+
+        signals = butterworth(signals, sampling_rate=1. / t_r,
+                              low_pass=low_pass, high_pass=high_pass)
+
     # Remove confounds
     if confounds is not None:
         confounds = _ensure_float(confounds)
         confounds = _standardize(confounds, normalize=standardize,
                                  detrend=detrend)
+
+        # Apply low- and high-pass filters to keep filters orthogonal
+        # (according to Lindquist et al. (2018))
+        if low_pass is not None or high_pass is not None:
+
+            confounds = butterworth(confounds, sampling_rate=1. / t_r,
+                                    low_pass=low_pass, high_pass=high_pass)
```

Comment:
We should filter before standardizing the confounds: the standardizing of confounds is useful for numerical stability.